usaspending_api/references/v2/views/filter_trees/tas_filter_tree.py

```diff
@@ -0,0 +1,94 @@
+from usaspending_api.common.helpers.agency_logic_helpers import cfo_presentation_order
+from usaspending_api.accounts.models import TreasuryAppropriationAccount, FederalAccount
+from usaspending_api.common.helpers.agency_logic_helpers import agency_from_identifiers
+from usaspending_api.references.v2.views.filter_trees.filter_tree import DEFAULT_CHILDREN, Node, FilterTree
+from usaspending_api.references.models import ToptierAgency
+
+
+class TASFilterTree(FilterTree):
+    def toptier_search(self):
```

Comment:
Are we supposed to be limiting these to only those agencies that have File C data?  Usually we don't want to show an agency if there's no way it can produce results.

Reply:
on line 10 the list of agency ids to search for is calculated from the TAS objects in order to ensure this.

Reply:
Having a TAS doesn't guarantee there's File C data for that TAS.

But that's sort of irrelevant.  I meant File D and edited my original comment accordingly.  My apologies.

Since this is an Advanced Search filter, it likely needs to only include those agencies, federal accounts, and treasury accounts that can be linked to File D data.  Maybe we don't needs this for minimum viability and can spin that off into its own ticket.  Might want to check with the PO.